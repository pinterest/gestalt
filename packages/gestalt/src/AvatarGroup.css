.overlay {
  height: 100%;

  /* Parent element
    Must be position=relative
    No z-index allowed
  */
  position: relative;
  width: 100%;
}

.overlay::after {
  /* Dark overlay is achieved overposing a pseudo-element with a 6% darker wash */
  background: var(--g-colorTransparentGray60);
  border-radius: 50%;
  content: "";
  display: block;
  height: 100%;
  position: absolute;
  top: 0;
  width: 100%;

  /* z-index required to position pseudo-elements in front of the parent */
  z-index: 1;
}

.translateX10 {
  transform: translateX(-10%);
}

html[dir="rtl"] .translateX10 {
  transform: translateX(10%);
}
