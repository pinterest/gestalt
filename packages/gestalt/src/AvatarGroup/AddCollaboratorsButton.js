// @flow strict
import { type Node as ReactNode } from 'react';
import { type BaseStackType } from './constants';
import HoverOverlay from './HoverOverlay';
import PositioningWrapper from './PositioningWrapper';
import AvatarFoundation from '../Avatar/Foundation';

export default function AvatarGroupAddCollaboratorsButton({
  hovered,
  pileCount,
  size,
}: BaseStackType): ReactNode {
  return (
    <PositioningWrapper index={pileCount - 1} pileCount={pileCount} size={size}>
      <HoverOverlay hovered={hovered} size={size}>
        <AvatarFoundation content="icon" outline />
      </HoverOverlay>
    </PositioningWrapper>
  );
}
