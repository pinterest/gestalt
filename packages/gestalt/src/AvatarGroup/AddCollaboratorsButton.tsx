import { BaseStackType } from './constants';
import HoverOverlay from './HoverOverlay';
import PositioningWrapper from './PositioningWrapper';
import AvatarFoundation from '../Avatar/Foundation';

type Props = BaseStackType & {
  isHovered: boolean;
  isPressed: boolean;
};

export default function AvatarGroupAddCollaboratorsButton({
  hovered,
  isHovered,
  isPressed,
  pileCount,
  size,
}: Props) {
  return (
    <PositioningWrapper index={pileCount - 1} pileCount={pileCount} size={size}>
      <HoverOverlay hovered={hovered} size={size}>
        <AvatarFoundation content="icon" isHovered={isHovered} isPressed={isPressed} outline />
      </HoverOverlay>
    </PositioningWrapper>
  );
}
