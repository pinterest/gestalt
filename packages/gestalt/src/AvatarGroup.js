// @flow strict
import {
  type AbstractComponent,
  forwardRef,
  Fragment,
  type Node as ReactNode,
  useState,
} from 'react';
import AddCollaboratorsButton from './AvatarGroup/AddCollaboratorsButton';
import CollaboratorAvatar from './AvatarGroup/CollaboratorAvatar';
import CollaboratorsCount from './AvatarGroup/CollaboratorsCount';
import Box from './Box';
import Flex from './Flex';
import TapArea from './TapArea';
import TapAreaLink from './TapAreaLink';

const MAX_COLLABORATOR_AVATARS = 3;

type UnionRefs = HTMLDivElement | HTMLAnchorElement;
type Props = {
  /**
   * Label for screen readers to announce AvatarGroup.
   *
   * See the [Accessibility guidelines](https://gestalt.pinterest.systems/web/avatargroup#Accessibility) for details on proper usage.
   */
  accessibilityLabel: string,
  /**
   * Specify the `id` of an associated element (or elements) whose contents or visibility are controlled by a component so that screen reader users can identify the relationship between elements. Optional with button-role component.
   *
   * See the [Accessibility guidelines](https://gestalt.pinterest.systems/web/avatargroup#Accessibility) for details on proper usage.
   */
  accessibilityControls?: string,
  /**
   * Indicate that a component hides or exposes collapsible components and expose whether they are currently expanded or collapsed. Optional with button-role component.
   *
   * See the [Accessibility guidelines](https://gestalt.pinterest.systems/web/avatargroup#Accessibility) for details on proper usage.
   */
  accessibilityExpanded?: boolean,
  /**
   * Indicate that a component controls the appearance of interactive popup elements, such as menu or dialog. Optional with button-role component.
   *
   * See the [Accessibility guidelines](https://gestalt.pinterest.systems/web/avatargroup#Accessibility) for details on proper usage.
   */
  accessibilityHaspopup?: boolean,
  /**
   * When supplied, it appends an `add` [icon](https://gestalt.pinterest.systems/web/icon) to the avatar pile as a call to action to the user. See [Best Practices](https://gestalt.pinterest.systems/web/avatargroup#Best-practices) for more info.
   */
  addCollaborators?: boolean,
  /**
   * The user group data. See the [collaborators display](https://gestalt.pinterest.systems/web/avatargroup#Collaborators-display) variant to learn more.
   */
  collaborators: $ReadOnlyArray<{
    name: string,
    src?: string,
  }>,
  /**
   * When supplied, wraps the component in a link, and directs users to the url when item is selected. See the [role](https://gestalt.pinterest.systems/web/avatargroup#Role) variant to learn more.
   */
  href?: string,
  /**
   * Callback fired when the component is clicked (pressed and released) with a mouse or keyboard. See the [role](https://gestalt.pinterest.systems/web/avatargroup#Role) variant to learn more and see [TapArea's `onTap`](https://gestalt.pinterest.systems/web/taparea#Props-onTap) for more info about `OnTapType`.
   */
  onClick?: ({
    event:
      | SyntheticMouseEvent<HTMLDivElement>
      | SyntheticKeyboardEvent<HTMLDivElement>
      | SyntheticMouseEvent<HTMLAnchorElement>
      | SyntheticKeyboardEvent<HTMLAnchorElement>,
    dangerouslyDisableOnNavigation: () => void,
  }) => void,
  /**
   * Forward the ref to the underlying div or anchor element. See the [role](https://gestalt.pinterest.systems/web/avatargroup#Role) variant to learn more.
   */
  ref?: UnionRefs, // eslint-disable-line react/no-unused-prop-types
  /**
   * Allows user interaction with the component. See the [role](https://gestalt.pinterest.systems/web/avatargroup#Role) variant to learn more.
   */
  role?: 'link' | 'button',
  /**
   * The maximum height of AvatarGroup. If size is `fit`, AvatarGroup will fill 100% of the parent container width. See the [fixed size](https://gestalt.pinterest.systems/web/avatargroup#Fixed-sizes) and [responsive size](https://gestalt.pinterest.systems/web/avatargroup#Responsive-sizing) variant to learn more.
   */
  size?: 'xs' | 'sm' | 'md' | 'fit',
};

/**
 * [AvatarGroup](https://gestalt.pinterest.systems/web/avatargroup) is used to both display a group of user avatars and, optionally, control actions related to the users group.
 *
 * ![AvatarGroup light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/AvatarGroup.spec.mjs-snapshots/AvatarGroup-chromium-darwin.png)
 * ![AvatarGroup dark mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/AvatarGroup-dark.spec.mjs-snapshots/AvatarGroup-dark-chromium-darwin.png)
 */
const AvatarGroupWithForwardRef: AbstractComponent<Props, UnionRefs> = forwardRef<Props, UnionRefs>(
  function AvatarGroup(
    {
      accessibilityLabel,
      accessibilityControls,
      accessibilityExpanded,
      accessibilityHaspopup,
      addCollaborators,
      collaborators,
      href,
      onClick,
      role,
      size = 'fit',
    }: Props,
    ref,
  ): ReactNode {
    const [hovered, setHovered] = useState(false);

    const isMdSize = size === 'md';

    const isFitSize = size === 'fit';

    const isMdOrFitSize = isMdSize || isFitSize;

    const isDisplayOnly = !role;

    const isAboveMaxCollaborators = collaborators.length > MAX_COLLABORATOR_AVATARS;

    const showCollaboratorsCount = isMdOrFitSize && isAboveMaxCollaborators;

    const showAddCollaboratorsButton =
      (isMdOrFitSize && !isDisplayOnly && addCollaborators) ?? false;

    const displayedCollaborators = collaborators.slice(
      0,
      isAboveMaxCollaborators && isMdOrFitSize ? 2 : MAX_COLLABORATOR_AVATARS,
    );

    const pileCount =
      displayedCollaborators.length +
      (showCollaboratorsCount ? 1 : 0) +
      (showAddCollaboratorsButton ? 1 : 0);

    const collaboratorStack = (
      <Fragment>
        {displayedCollaborators.map(({ src, name }, index) => (
          <CollaboratorAvatar
            // eslint-disable-next-line react/no-array-index-key
            key={`collaboratorStack-${name}-${index}`}
            hovered={hovered}
            index={index}
            name={name}
            pileCount={pileCount}
            size={size}
            src={src || ''}
          />
        ))}
        {showCollaboratorsCount && (
          <CollaboratorsCount
            key={`collaboratorStack-count-${collaborators.length}`}
            count={collaborators.length - 2}
            hovered={hovered}
            pileCount={pileCount}
            showAddCollaboratorsButton={showAddCollaboratorsButton}
            size={size}
          />
        )}
        {showAddCollaboratorsButton && (
          <AddCollaboratorsButton
            key={`collaboratorStack-addButton-${collaborators.length}`}
            hovered={hovered}
            pileCount={pileCount}
            size={size}
          />
        )}
      </Fragment>
    );

    const avatarGroupStack = (
      <Box
        aria-label={isDisplayOnly ? accessibilityLabel : undefined}
        dangerouslySetInlineStyle={{ __style: { isolation: 'isolate' } }}
        position={isFitSize ? 'relative' : 'static'}
      >
        {isFitSize ? collaboratorStack : <Flex>{collaboratorStack}</Flex>}
      </Box>
    );

    if (role === 'link' && href) {
      return (
        <TapAreaLink
          ref={ref}
          accessibilityLabel={accessibilityLabel}
          fullWidth={false}
          href={href}
          onMouseEnter={() => setHovered(true)}
          onMouseLeave={() => setHovered(false)}
          onTap={({ event, dangerouslyDisableOnNavigation }) =>
            onClick?.({ event, dangerouslyDisableOnNavigation })
          }
          rounding="pill"
          tapStyle="compress"
        >
          {avatarGroupStack}
        </TapAreaLink>
      );
    }

    if (role === 'button' && onClick) {
      return (
        <TapArea
          ref={ref}
          accessibilityControls={accessibilityControls}
          accessibilityExpanded={accessibilityExpanded}
          accessibilityHaspopup={accessibilityHaspopup}
          accessibilityLabel={accessibilityLabel}
          fullWidth={false}
          onMouseEnter={() => setHovered(true)}
          onMouseLeave={() => setHovered(false)}
          // $FlowExpectedError[prop-missing]
          onTap={({ event }) => onClick({ event })}
          rounding="pill"
          tapStyle="compress"
        >
          {avatarGroupStack}
        </TapArea>
      );
    }

    // Display-only role
    return avatarGroupStack;
  },
);

AvatarGroupWithForwardRef.displayName = 'AvatarGroup';

export default AvatarGroupWithForwardRef;
