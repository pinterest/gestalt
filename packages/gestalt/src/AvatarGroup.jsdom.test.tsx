import { ComponentProps, createRef } from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import AvatarGroup from './AvatarGroup';
import Box from './Box';

describe('AvatarGroup', () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });

  const renderCmp = ({
    addCollaborators = false,
    collaborators,
    href,
    onClick,
    role,
    size = 'md',
    avatarRef,
  }: {
    addCollaborators?: ComponentProps<typeof AvatarGroup>['addCollaborators'];
    collaborators: ComponentProps<typeof AvatarGroup>['collaborators'];
    href?: ComponentProps<typeof AvatarGroup>['href'];
    onClick?: ComponentProps<typeof AvatarGroup>['onClick'];
    role?: ComponentProps<typeof AvatarGroup>['role'];
    size?: ComponentProps<typeof AvatarGroup>['size'];
    avatarRef?: ComponentProps<typeof AvatarGroup>['ref'];
  }) =>
    render(
      size === 'fit' ? (
        <Box width={200}>
          <AvatarGroup
            ref={avatarRef}
            accessibilityLabel="test"
            addCollaborators={addCollaborators}
            collaborators={collaborators}
            href={href}
            onClick={onClick}
            role={role}
            size={size}
          />
        </Box>
      ) : (
        <AvatarGroup
          ref={avatarRef}
          accessibilityLabel="test"
          addCollaborators={addCollaborators}
          collaborators={collaborators}
          href={href}
          onClick={onClick}
          role={role}
          size={size}
        />
      ),
    );

  it('renders xs display-only AvatarGroup with image', () => {
    const { baseElement } = renderCmp({
      collaborators: [
        {
          name: 'Keerthi',
          src: 'https://i.pinimg.com/originals/bf/bc/27/bfbc27685d81eb9a8f65c201ea661f0e.jpg',
        },
      ],
      size: 'xs',
    });
    expect(baseElement).toMatchSnapshot();
  });

  it('renders sm display-only AvatarGroup with image', () => {
    const { baseElement } = renderCmp({
      collaborators: [
        {
          name: 'Keerthi',
          src: 'https://i.pinimg.com/originals/bf/bc/27/bfbc27685d81eb9a8f65c201ea661f0e.jpg',
        },
      ],
      size: 'sm',
    });
    expect(baseElement).toMatchSnapshot();
  });

  it('renders md display-only AvatarGroup with image', () => {
    const { baseElement } = renderCmp({
      collaborators: [
        {
          name: 'Keerthi',
          src: 'https://i.pinimg.com/originals/bf/bc/27/bfbc27685d81eb9a8f65c201ea661f0e.jpg',
        },
      ],
    });
    expect(baseElement).toMatchSnapshot();
  });

  it('renders fit display-only AvatarGroup with image', () => {
    const { baseElement } = renderCmp({
      collaborators: [
        {
          name: 'Keerthi',
          src: 'https://i.pinimg.com/originals/bf/bc/27/bfbc27685d81eb9a8f65c201ea661f0e.jpg',
        },
      ],
      size: 'fit',
      role: 'button',
      onClick: jest.fn(),
    });
    expect(baseElement).toMatchSnapshot();
  });

  it('renders display-only AvatarGroup without image', () => {
    const { baseElement } = renderCmp({
      collaborators: [
        {
          name: 'Keerthi',
        },
      ],
    });
    expect(baseElement).toMatchSnapshot();
  });

  it('renders display-only AvatarGroup with counter', () => {
    const { baseElement } = renderCmp({
      collaborators: Array(5).fill({
        name: 'Keerthi',
        src: 'https://i.pinimg.com/originals/bf/bc/27/bfbc27685d81eb9a8f65c201ea661f0e.jpg',
      }),
    });
    expect(screen.getByText('3')).toBeVisible();
    expect(baseElement).toMatchSnapshot();
  });

  it('renders display-only 99+ counter over 101 collaborators', () => {
    renderCmp({
      collaborators: Array(102).fill({
        name: 'Keerthi',
        src: 'https://i.pinimg.com/originals/bf/bc/27/bfbc27685d81eb9a8f65c201ea661f0e.jpg',
      }),
    });
    expect(screen.queryAllByAltText('Keerthi')).toHaveLength(2);
    expect(screen.getByText('99+')).toBeVisible();
  });

  it('renders AvatarGroup button with addCollaborators button', () => {
    const { baseElement } = renderCmp({
      collaborators: [
        {
          name: 'Keerthi',
          src: 'https://i.pinimg.com/originals/bf/bc/27/bfbc27685d81eb9a8f65c201ea661f0e.jpg',
        },
      ],
      addCollaborators: true,
      role: 'button',
    });
    expect(baseElement).toMatchSnapshot();
  });

  it('renders AvatarGroup link with addCollaborators button', () => {
    const { baseElement } = renderCmp({
      collaborators: [
        {
          name: 'Keerthi',
          src: 'https://i.pinimg.com/originals/bf/bc/27/bfbc27685d81eb9a8f65c201ea661f0e.jpg',
        },
      ],
      addCollaborators: true,
      role: 'link',
      href: 'www.pinterest.com',
    });
    expect(baseElement).toMatchSnapshot();
  });

  it('should render clickable TapArea with button role and ref', () => {
    const onClickMock = jest.fn<
      [
        {
          dangerouslyDisableOnNavigation: () => void;
          event:
            | React.MouseEvent<HTMLDivElement>
            | React.KeyboardEvent<HTMLDivElement>
            | React.MouseEvent<HTMLAnchorElement>
            | React.KeyboardEvent<HTMLAnchorElement>;
        },
      ],
      // @ts-expect-error - TS2344 - Type 'undefined' does not satisfy the constraint 'any[]'.
      undefined
    >();
    const ref = createRef<HTMLDivElement>();
    renderCmp({
      collaborators: [
        {
          name: 'Keerthi',
          src: 'https://i.pinimg.com/originals/bf/bc/27/bfbc27685d81eb9a8f65c201ea661f0e.jpg',
        },
      ],
      onClick: onClickMock,
      role: 'button',
      avatarRef: ref,
    });
    expect(ref.current instanceof HTMLDivElement).toEqual(true);
    fireEvent.click(screen.getByAltText('Keerthi'));
    expect(onClickMock).toHaveBeenCalled();
  });

  it('should render clickable TapArea with link role and ref', () => {
    const onClickMock = jest.fn<
      [
        {
          dangerouslyDisableOnNavigation: () => void;
          event:
            | React.MouseEvent<HTMLDivElement>
            | React.KeyboardEvent<HTMLDivElement>
            | React.MouseEvent<HTMLAnchorElement>
            | React.KeyboardEvent<HTMLAnchorElement>;
        },
      ],
      // @ts-expect-error - TS2344 - Type 'undefined' does not satisfy the constraint 'any[]'.
      undefined
    >();
    const ref = createRef<HTMLDivElement>();
    renderCmp({
      collaborators: [
        {
          name: 'Keerthi',
          src: 'https://i.pinimg.com/originals/bf/bc/27/bfbc27685d81eb9a8f65c201ea661f0e.jpg',
        },
      ],
      onClick: onClickMock,
      role: 'link',
      href: 'http://www.pinterest.com',
      avatarRef: ref,
    });
    expect(ref.current instanceof HTMLAnchorElement).toEqual(true);
    fireEvent.click(screen.getByAltText('Keerthi'));
    expect(onClickMock).toHaveBeenCalled();
  });

  it('should not call onClick when clicked on display-only role', () => {
    const onClickMock = jest.fn<
      [
        {
          dangerouslyDisableOnNavigation: () => void;
          event:
            | React.MouseEvent<HTMLDivElement>
            | React.KeyboardEvent<HTMLDivElement>
            | React.MouseEvent<HTMLAnchorElement>
            | React.KeyboardEvent<HTMLAnchorElement>;
        },
      ],
      // @ts-expect-error - TS2344 - Type 'undefined' does not satisfy the constraint 'any[]'.
      undefined
    >();
    renderCmp({
      collaborators: [
        {
          name: 'Keerthi',
          src: 'https://i.pinimg.com/originals/bf/bc/27/bfbc27685d81eb9a8f65c201ea661f0e.jpg',
        },
      ],
      onClick: onClickMock,
    });
    fireEvent.click(screen.getByAltText('Keerthi'));
    expect(onClickMock).not.toHaveBeenCalled();
  });
});
