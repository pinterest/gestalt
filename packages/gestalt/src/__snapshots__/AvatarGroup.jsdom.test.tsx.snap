// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarGroup renders AvatarGroup button with addCollaborators button 1`] = `
<body>
  <div>
    <div
      class="box"
      style="isolation: isolate;"
    >
      <div
        class="Flex rowGap0 columnGap0 xsDirectionRow"
      >
        <div
          aria-hidden="true"
          class="box"
          style="margin-inline-start: 0px;"
        >
          <div
            class="box relative"
            style="height: 48px; width: 48px; z-index: 1;"
          >
            <div
              class=""
            >
              <div
                class="box circle relative"
                data-test-id="gestalt-avatar-svg"
                style="box-shadow: 0 0 0 1px var(--color-border-avatar); height: 48px; width: 48px;"
              >
                <div
                  class="Mask circle willChangeTransform"
                >
                  <div
                    class="box relative"
                    style="padding-bottom: 100%;"
                  >
                    <img
                      alt="Keerthi"
                      class="img"
                      fetchpriority="auto"
                      loading="auto"
                      src="https://i.ibb.co/ZfCZrY8/keerthi.jpg"
                    />
                  </div>
                  <div
                    class="wash"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="box"
          style="margin-inline-start: -16px;"
        >
          <div
            class="box relative"
            style="height: 48px; width: 48px; z-index: 1;"
          >
            <div
              class=""
            >
              <div
                class="box circle relative secondary"
                style="padding-bottom: 100%; box-shadow: 0 0 0 1px var(--color-border-avatar);"
              >
                <div
                  class="absolute bottom0 box justifyCenter left0 right0 top0 xsDisplayFlex"
                >
                  <svg
                    class="icon text"
                    preserveAspectRatio="xMidYMid meet"
                    role="img"
                    version="1.1"
                    viewBox="0 0 24 24"
                    width="41.66666666666667%"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <title>
                      Icon
                    </title>
                    <path
                      d="test-file-stub"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`AvatarGroup renders AvatarGroup link with addCollaborators button 1`] = `
<body>
  <div>
    <a
      aria-label="test"
      class="noOutline inheritColor noUnderline tapTransition pill accessibilityOutline block pointer"
      href="www.pinterest.com"
      rel=""
      tabindex="0"
    >
      <div
        class="box"
        style="isolation: isolate;"
      >
        <div
          class="Flex rowGap0 columnGap0 xsDirectionRow"
        >
          <div
            aria-hidden="true"
            class="box"
            style="margin-inline-start: 0px;"
          >
            <div
              class="box relative"
              style="height: 48px; width: 48px; z-index: 1;"
            >
              <div
                class=""
              >
                <div
                  class="box circle relative"
                  data-test-id="gestalt-avatar-svg"
                  style="box-shadow: 0 0 0 1px var(--color-border-avatar); height: 48px; width: 48px;"
                >
                  <div
                    class="Mask circle willChangeTransform"
                  >
                    <div
                      class="box relative"
                      style="padding-bottom: 100%;"
                    >
                      <img
                        alt="Keerthi"
                        class="img"
                        fetchpriority="auto"
                        loading="auto"
                        src="https://i.ibb.co/ZfCZrY8/keerthi.jpg"
                      />
                    </div>
                    <div
                      class="wash"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="box"
            style="margin-inline-start: -16px;"
          >
            <div
              class="box relative"
              style="height: 48px; width: 48px; z-index: 1;"
            >
              <div
                class=""
              >
                <div
                  class="box circle relative secondary"
                  style="padding-bottom: 100%; box-shadow: 0 0 0 1px var(--color-border-avatar);"
                >
                  <div
                    class="absolute bottom0 box justifyCenter left0 right0 top0 xsDisplayFlex"
                  >
                    <svg
                      class="icon text"
                      preserveAspectRatio="xMidYMid meet"
                      role="img"
                      version="1.1"
                      viewBox="0 0 24 24"
                      width="41.66666666666667%"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <title>
                        Icon
                      </title>
                      <path
                        d="test-file-stub"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </a>
  </div>
</body>
`;

exports[`AvatarGroup renders display-only AvatarGroup with counter 1`] = `
<body>
  <div>
    <div
      aria-label="test"
      class="box"
      style="isolation: isolate;"
    >
      <div
        class="Flex rowGap0 columnGap0 xsDirectionRow"
      >
        <div
          aria-hidden="true"
          class="box"
          style="margin-inline-start: 0px;"
        >
          <div
            class="box relative"
            style="height: 48px; width: 48px; z-index: 1;"
          >
            <div
              class=""
            >
              <div
                class="box circle relative"
                data-test-id="gestalt-avatar-svg"
                style="box-shadow: 0 0 0 1px var(--color-border-avatar); height: 48px; width: 48px;"
              >
                <div
                  class="Mask circle willChangeTransform"
                >
                  <div
                    class="box relative"
                    style="padding-bottom: 100%;"
                  >
                    <img
                      alt="Keerthi"
                      class="img"
                      fetchpriority="auto"
                      loading="auto"
                      src="https://i.ibb.co/ZfCZrY8/keerthi.jpg"
                    />
                  </div>
                  <div
                    class="wash"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="box"
          style="margin-inline-start: -16px;"
        >
          <div
            class="box relative"
            style="height: 48px; width: 48px; z-index: 1;"
          >
            <div
              class=""
            >
              <div
                class="box circle relative"
                data-test-id="gestalt-avatar-svg"
                style="box-shadow: 0 0 0 1px var(--color-border-avatar); height: 48px; width: 48px;"
              >
                <div
                  class="Mask circle willChangeTransform"
                >
                  <div
                    class="box relative"
                    style="padding-bottom: 100%;"
                  >
                    <img
                      alt="Keerthi"
                      class="img"
                      fetchpriority="auto"
                      loading="auto"
                      src="https://i.ibb.co/ZfCZrY8/keerthi.jpg"
                    />
                  </div>
                  <div
                    class="wash"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="box"
          style="margin-inline-start: -16px;"
        >
          <div
            class="box relative"
            style="height: 48px; width: 48px; z-index: 1;"
          >
            <div
              class=""
            >
              <div
                class="box circle relative secondary"
                style="padding-bottom: 100%; box-shadow: 0 0 0 1px var(--color-border-avatar);"
              >
                <div
                  class="absolute bottom0 box justifyCenter left0 right0 top0 xsDisplayFlex"
                >
                  <svg
                    preserveAspectRatio="xMidYMid meet"
                    version="1.1"
                    viewBox="-50 -50 100 100"
                    width="100%"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <text
                      class="antialiased sansSerif fontWeightSemiBold "
                      dy="0.35em"
                      fill="var(--color-text-default)"
                      font-size="40px"
                      text-anchor="middle"
                    >
                      3
                    </text>
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`AvatarGroup renders display-only AvatarGroup without image 1`] = `
<body>
  <div>
    <div
      aria-label="test"
      class="box"
      style="isolation: isolate;"
    >
      <div
        class="Flex rowGap0 columnGap0 xsDirectionRow"
      >
        <div
          aria-hidden="true"
          class="box"
          style="margin-inline-start: 0px;"
        >
          <div
            class="box relative"
            style="height: 48px; width: 48px; z-index: 1;"
          >
            <div
              class=""
            >
              <div
                class="box circle relative"
                data-test-id="gestalt-avatar-svg"
                style="box-shadow: 0 0 0 1px var(--color-border-avatar); height: 48px; width: 48px;"
              >
                <div
                  class="box circle relative secondary"
                  style="padding-bottom: 100%;"
                >
                  <div
                    class="absolute bottom0 box justifyCenter left0 right0 top0 xsDisplayFlex"
                  >
                    <svg
                      preserveAspectRatio="xMidYMid meet"
                      version="1.1"
                      viewBox="-50 -50 100 100"
                      width="100%"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <title>
                        Keerthi
                      </title>
                      <text
                        class="antialiased sansSerif fontWeightSemiBold "
                        dy="0.35em"
                        fill="var(--color-text-default)"
                        font-size="40px"
                        text-anchor="middle"
                      >
                        K
                      </text>
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`AvatarGroup renders fit display-only AvatarGroup with image 1`] = `
<body>
  <div>
    <div
      class="box"
      style="width: 200px;"
    >
      <div
        aria-disabled="false"
        aria-label="test"
        class="tapTransition pill accessibilityOutline pointer"
        role="button"
        tabindex="0"
      >
        <div
          class="box relative"
          style="isolation: isolate;"
        >
          <div
            aria-hidden="true"
            class="absolute box"
            style="margin-inline-start: 0%; top: 0px; width: 100%;"
          >
            <div
              class="box relative"
              style="height: 100%; width: 100%; z-index: 1;"
            >
              <div
                class=""
              >
                <div
                  class="box circle relative"
                  data-test-id="gestalt-avatar-svg"
                  style="box-shadow: 0 0 0 1px var(--color-border-avatar); width: 100%;"
                >
                  <div
                    class="Mask circle willChangeTransform"
                  >
                    <div
                      class="box relative"
                      style="padding-bottom: 100%;"
                    >
                      <img
                        alt="Keerthi"
                        class="img"
                        fetchpriority="auto"
                        loading="auto"
                        src="https://i.ibb.co/ZfCZrY8/keerthi.jpg"
                      />
                    </div>
                    <div
                      class="wash"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`AvatarGroup renders md display-only AvatarGroup with image 1`] = `
<body>
  <div>
    <div
      aria-label="test"
      class="box"
      style="isolation: isolate;"
    >
      <div
        class="Flex rowGap0 columnGap0 xsDirectionRow"
      >
        <div
          aria-hidden="true"
          class="box"
          style="margin-inline-start: 0px;"
        >
          <div
            class="box relative"
            style="height: 48px; width: 48px; z-index: 1;"
          >
            <div
              class=""
            >
              <div
                class="box circle relative"
                data-test-id="gestalt-avatar-svg"
                style="box-shadow: 0 0 0 1px var(--color-border-avatar); height: 48px; width: 48px;"
              >
                <div
                  class="Mask circle willChangeTransform"
                >
                  <div
                    class="box relative"
                    style="padding-bottom: 100%;"
                  >
                    <img
                      alt="Keerthi"
                      class="img"
                      fetchpriority="auto"
                      loading="auto"
                      src="https://i.ibb.co/ZfCZrY8/keerthi.jpg"
                    />
                  </div>
                  <div
                    class="wash"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`AvatarGroup renders sm display-only AvatarGroup with image 1`] = `
<body>
  <div>
    <div
      aria-label="test"
      class="box"
      style="isolation: isolate;"
    >
      <div
        class="Flex rowGap0 columnGap0 xsDirectionRow"
      >
        <div
          aria-hidden="true"
          class="box"
          style="margin-inline-start: 0px;"
        >
          <div
            class="box relative"
            style="height: 32px; width: 32px; z-index: 1;"
          >
            <div
              class=""
            >
              <div
                class="box circle relative"
                data-test-id="gestalt-avatar-svg"
                style="box-shadow: 0 0 0 1px var(--color-border-avatar); height: 32px; width: 32px;"
              >
                <div
                  class="Mask circle willChangeTransform"
                >
                  <div
                    class="box relative"
                    style="padding-bottom: 100%;"
                  >
                    <img
                      alt="Keerthi"
                      class="img"
                      fetchpriority="auto"
                      loading="auto"
                      src="https://i.ibb.co/ZfCZrY8/keerthi.jpg"
                    />
                  </div>
                  <div
                    class="wash"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`AvatarGroup renders xs display-only AvatarGroup with image 1`] = `
<body>
  <div>
    <div
      aria-label="test"
      class="box"
      style="isolation: isolate;"
    >
      <div
        class="Flex rowGap0 columnGap0 xsDirectionRow"
      >
        <div
          aria-hidden="true"
          class="box"
          style="margin-inline-start: 0px;"
        >
          <div
            class="box relative"
            style="height: 24px; width: 24px; z-index: 1;"
          >
            <div
              class=""
            >
              <div
                class="box circle relative"
                data-test-id="gestalt-avatar-svg"
                style="box-shadow: 0 0 0 1px var(--color-border-avatar); height: 24px; width: 24px;"
              >
                <div
                  class="Mask circle willChangeTransform"
                >
                  <div
                    class="box relative"
                    style="padding-bottom: 100%;"
                  >
                    <img
                      alt="Keerthi"
                      class="img"
                      fetchpriority="auto"
                      loading="auto"
                      src="https://i.ibb.co/ZfCZrY8/keerthi.jpg"
                    />
                  </div>
                  <div
                    class="wash"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
